crates/typst/src/lib.rs

```diff
@@ -231,6 +231,51 @@ pub trait World {
     }
 }
 
+macro_rules! delegate_for_ptr {
+    ($W:ident for $ptr:ty) => {
+        impl<$W: World> World for $ptr
+        where
+            $ptr: Deref<Target = $W>,
+        {
+            fn library(&self) -> &Prehashed<Library> {
+                self.deref().library()
+            }
+
+            fn book(&self) -> &Prehashed<FontBook> {
+                self.deref().book()
+            }
+
+            fn main(&self) -> Source {
+                self.deref().main()
+            }
+
+            fn source(&self, id: FileId) -> FileResult<Source> {
+                self.deref().source(id)
+            }
+
+            fn file(&self, id: FileId) -> FileResult<Bytes> {
+                self.deref().file(id)
+            }
+
+            fn font(&self, index: usize) -> Option<Font> {
+                self.deref().font(index)
+            }
+
+            fn today(&self, offset: Option<i64>) -> Option<Datetime> {
+                self.deref().today(offset)
+            }
+
+            fn packages(&self) -> &[(PackageSpec, Option<EcoString>)] {
+                self.deref().packages()
+            }
+        }
+    };
+}
+
+delegate_for_ptr!(W for std::boxed::Box<W>);
+delegate_for_ptr!(W for std::sync::Arc<W>);
+delegate_for_ptr!(W for std::rc::Rc<W>);
```

Comment:
I'd prefer not to have this `Rc` impl in the first place since I want to add `Send + Sync` bounds to `World` in the future (for multi-threaded compilation).